### PR TITLE
Updating logging and metrics to restart api, ha and controllers when …

### PR DESCRIPTION
**Pull request by ewolinetz**
…updating master config

Ideally we would have a common role that contains handlers that we could make a dependency, but this should work for now.

Also noticed a bug while testing in logging regarding updating a route (fix in oc_apply.yaml)

cc @dgoodwin

### CODE REVIEW

**Comment by ewolinetz** [comment]
Should address #3997

**Comment by ewolinetz** [comment]
Handler taken from https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_master/handlers/main.yml

File: roles/openshift_logging/tasks/oc_apply.yaml [hunk 1]
```diff
@@ -49,4 +49,4 @@
     failed_when: "'error' in generation_apply.stderr"
     changed_when: generation_apply.rc == 0
     when: "'field is immutable' in generation_apply.stderr"
-  when: file_content.kind == "Service"
+  when: file_content.kind in ["Service", "Route"]
```
Comment (sdodson) [inline roles/openshift_logging/tasks/oc_apply.yaml]: unrelated changes?

File: roles/openshift_metrics/tasks/update_master_config.yaml [hunk 2]
```diff
@@ -4,6 +4,9 @@
     dest: "{{ openshift.common.config_base }}/master/master-config.yaml"
     yaml_key: assetConfig.metricsPublicURL
     yaml_value: "https://{{ openshift_metrics_hawkular_hostname}}/hawkular/metrics"
-  notify: restart master
+  notify:
+  - restart master
+  - restart master api
```
Comment (dgoodwin) [inline roles/openshift_metrics/tasks/update_master_config.yaml]: Suggest dropping "HA" into these handler descriptions just for readability

File: roles/openshift_logging/tasks/oc_apply.yaml (hunk 1 above)
Comment (ewolinetz) [inline roles/openshift_logging/tasks/oc_apply.yaml]: its to address a bug I came across while doing some quick local tests... changing the kibana hostname after deploying initially ran into the same issue we see with services

File: roles/openshift_metrics/tasks/update_master_config.yaml (hunk 2 above)
Comment (ewolinetz) [inline roles/openshift_metrics/tasks/update_master_config.yaml]: i was thinking since its currently copy/pasta, just to make it easier to rip it all out later, i'd keep the name... but i'm open to it

**Comment by ewolinetz** [comment]
aos-ci-test

**Comment by ewolinetz** [comment]
aos-ci-test

**Comment by ewolinetz** [comment]
aos-ci-test

**Comment by ewolinetz** [comment]
aos-ci-test

**Comment by ewolinetz** [comment]
aos-ci-test

**Comment by ewolinetz** [comment]
[merge]
